clients/extension/src/background/index.ts

```diff
@@ -78,14 +77,17 @@ const handleOpenPanel = (name: string): Promise<number> => {
   })
 }
 
-const handleProvider = (chain: Chain, update?: boolean) => {
+const handleProvider = (chain: Chain, update?: boolean): boolean => {
   const rpc = chainRpcUrl[chain]
-  if (!rpc) return
-  if (update && rpcProvider) {
-    rpcProvider = new JsonRpcProvider(rpc)
-    return
+  if (rpc) {
```

Comment:
Honestly , I like the previously one better ,  nest `if` is always harder to read

Reply:
Resolved